File: Solutions/456. 132 Pattern/456. 132 Pattern.go
func find132pattern(nums []int) bool {
    z:=math.MinInt64
    var stk []int
    for i:=len(nums)-1;i>=0;i-=1{
        if nums[i]<z{
            return true
        }else{
            for len(stk)>0 && nums[i]>stk[len(stk)-1]{
                z=stk[len(stk)-1];stk=stk[:len(stk)-1]
            }
            stk=append(stk,nums[i])
        }
    }
    return false
}

File: Solutions/456. 132 Pattern/456. 132 Pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
        int z=INT_MIN;
        stack<int> stk;
        for(int i=nums.size()-1;i>=0;--i){
            if(nums[i]<z){
                return true;
            }else{
                while(stk.size() and nums[i]>stk.top()){
                    z=stk.top();stk.pop();
                }
                stk.push(nums[i]);
            }
        }
        return false;
    }
};